pkg/operator/ceph/cluster/cluster.go

```diff
@@ -347,6 +354,26 @@ func (c *ClusterController) preClusterStartValidation(cluster *cluster, clusterO
 	if len(cluster.Spec.Storage.Directories) != 0 {
 		logger.Warning("running osds on directory is not supported anymore, use devices instead.")
 	}
+	if cluster.Spec.IsStretchCluster() {
+		if len(cluster.Spec.Mon.StretchCluster.Zones) != 3 {
```

Comment:
I believe this block could have its own method so the main `preClusterStartValidation` does not get too long.